Can't build using MSYS2/MinGW-w64

I tried using both "MSYS Makefiles" and "Unix Makefiles" generators.

I get the following error:

```
$ make
Scanning dependencies of target libnpkStatic
[  1%] Building C object libnpk/CMakeFiles/libnpkStatic.dir/src/npk.c.obj
In file included from C:/msys64/home/Guillaume/npk/libnpk/include/npk.h:11:0,
                 from C:/msys64/home/Guillaume/npk/libnpk/src/npk.c:12:
C:/msys64/home/Guillaume/npk/libnpk/include/npk_conf.h:41:15: error: conflicting types for 'mode_t'
   typedef int mode_t;
               ^~~~~~
In file included from C:/msys64/mingw32/i686-w64-mingw32/include/sys/stat.h:26:0,
                 from C:/msys64/home/Guillaume/npk/libnpk/src/npk.c:11:
C:/msys64/mingw32/i686-w64-mingw32/include/sys/types.h:77:17: note: previous declaration of 'mode_t' was here
 typedef _mode_t mode_t;
                 ^~~~~~
make[2]: *** [libnpk/CMakeFiles/libnpkStatic.dir/build.make:64: libnpk/CMakeFiles/libnpkStatic.dir/src/npk.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:90: libnpk/CMakeFiles/libnpkStatic.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
```

Though I use MSYS2 64-bit, I tried to build using the 32-bit version of mingw-w64 (since the 64-bit version gives a different error).

Don't hesitate to ask if you need more info.

--- a/libnpk/CMakeLists.txt
+++ b/libnpk/CMakeLists.txt
@@ -34,11 +34,9 @@ set( NPK_SRCS_DEV_MODE
     ./src/npk_dev.c
 )
 
-if( WIN32 )
-    if( NOT CMAKE_CL_64 )
-        add_definitions( -D_USE_32BIT_TIME_T )
-    endif( NOT CMAKE_CL_64 )
-endif( WIN32 )
+if( NOT CMAKE_SIZEOF_VOID_P EQUAL 8 )
+    add_definitions( -D_USE_32BIT_TIME_T )
+endif( NOT CMAKE_SIZEOF_VOID_P EQUAL 8 )
 
 if( USE_ZLIB_PREFIX )
     add_definitions( -DZ_PREFIX )

--- a/libnpk/include/npk_conf.h
+++ b/libnpk/include/npk_conf.h
@@ -38,7 +38,9 @@
 
 #if defined(_WIN32) || defined(_WIN64)
     #define NPK_PLATFORM_WINDOWS
- 	typedef int mode_t;
+    #ifndef _MODE_T_
+        typedef int mode_t;
+    #endif
 #elif defined(linux) || defined(__linux) || defined(__linux__)
     #define NPK_PLATFORM_LINUX
 #elif defined(__FreeBSD__)

--- a/libnpk/tests/testutil.h
+++ b/libnpk/tests/testutil.h
@@ -4,6 +4,7 @@
 
 #include <iostream>
 #include <fcntl.h>
+#include <malloc.h>
 #if defined( WIN32 )
 #include <io.h>
 #pragma warning ( disable : 4819 )

